source/util/hex_float.h

```diff
@@ -1059,9 +1059,13 @@ std::istream& operator>>(std::istream& is, HexFloat<T, Traits>& value) {
     } else if (::isdigit(next_char)) {
       seen_written_exponent_digits = true;
       // Hex-floats express their exponent as decimal.
-      written_exponent = static_cast<int_type>(written_exponent * 10);
-      written_exponent =
-          static_cast<int_type>(written_exponent + (next_char - '0'));
+      int_type digit = (static_cast<int_type>(next_char) - '0');
+      if (written_exponent > (HF::max_exponent - digit) / 10) {
```

Comment:
Hm.   When the exponent is large negative, this will bail out, when it would not have done so before.

E.g.    0x1p-129  is a test case that now fails. But it should succeed but underflow.

(Test case:  Float32Tests/HexFloatTest.DecodeCorrectly/28 in spirv-tools-test_spirv_unit_tests)